src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperUseageDescriptorFactory.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if !DNXCORE50 // Cannot accurately resolve the location of the documentation XML file in coreclr.
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNet.Razor.TagHelpers;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public static class TagHelperUseageDescriptorFactory
+    {
+        private static readonly Dictionary<string, XmlDocumentationProvider> XmlDocumentationProviderCache =
+            new Dictionary<string, XmlDocumentationProvider>(StringComparer.Ordinal);
```

Comment:
This prevents Razor parsing from going to the file system. We've always pre-emptively added caches for file system like behaviors.